[Nixx162] iP

Nixx162 opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Nixx162 committed:
feat: add welcome and exit message

Nixx162 committed:
feat: add echo feature

Nixx162 committed:
feat: add tasklist feature

Nixx162 committed:
refactor: implement Parser class for commands

Nixx162 committed:
feat: add marking as done feature

Nixx162 committed:
style: add some Italian flair

Nixx162 committed:
feat: add diferent task types and error messages

Nixx162 committed:
feat: add delete feature

Nixx162 committed:
fix: fix a parsing bug

Nixx162 committed:
test: add test script and some test cases

Nixx162 committed:
feat: add list storage capability

Nixx162 committed:
Merge branch 'branch-Level-7'

Nixx162 committed:
feat: Add support for date and time

Nixx162 committed:
Merge branch 'branch-Level-8'

Nixx162 committed:
Merge branch 'add-gradle-support'

tohpinren commented on src/main/java/Pardiyem.java:
```diff
@@ -0,0 +1,41 @@
+                Command c = Parser.parseCommand(fullCommand);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+            } finally {
```
Should there be a catch block for the try block?

tohpinren commented on src/main/java/pardiyem/command/AddCommand.java:
```diff
@@ -0,0 +1,47 @@
+            ArrayList<String> args = Event.parseDesc(desc);
+            toAdd = new Event(args.get(0), args.get(1), args.get(2));
+            break;
+        }
```
Should there be a default case? You could use the default branch for the intended default action or if there is no default action, you can use the default branch to detect errors.

tohpinren commented on src/main/java/pardiyem/parser/Parser.java:
```diff
@@ -0,0 +1,43 @@
+            command =  in.substring(0, x);
+        }
+        switch (command) {
+            case "bye":
```
Should there be indentation for case clauses?

tohpinren commented on src/main/java/pardiyem/parser/Parser.java:
```diff
@@ -0,0 +1,43 @@
+                return new AddCommand(in.substring(5).trim(), 3);
+            case "delete":
+                return new DeleteCommand(in.substring(6).trim());
+            default:
```
I like the use of default case here to catch errors.

tohpinren commented on src/main/java/pardiyem/storage/Storage.java:
```diff
@@ -0,0 +1,80 @@
+            return new Event(line.substring(7, indFrom - 2), 
+                    line.substring(indFrom + 6, indTo - 1),
+                    line.substring(indTo + 4, line.length() - 1),
+                    isDone);
```
This line is a little long and confusing maybe you could split the arguments into variables first?

tohpinren reviewed (commented):
Overall, your code was pretty readable and you follow almost all of the coding standards. Some empty lines could be used to split up long blocks of code for better readability. I noted a few minor problems regarding code quality guidelines.

Nixx162 committed:
test: Add several JUnit test classes

Nixx162 committed:
chore: Update gradle for checkstyle

Nixx162 committed:
test: Add equals method to classes to facilitate testing

Nixx162 committed:
refactor: Add new constructor to Deadline and Event class for better abstraction

Nixx162 committed:
docs: Add javadoc comment to some public methods and classes

Nixx162 committed:
style: Change some lines to abide by the given coding standard

Nixx162 committed:
style: Change some lines on the test classes to fit coding standards

Nixx162 committed:
feat: Add task-finding capability to the chatbot

Nixx162 committed:
Merge branch 'branch-A-JavaDoc'

Nixx162 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/pardiyem/command/ByeCommand.java
#	src/main/java/pardiyem/storage/Storage.java

Nixx162 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/pardiyem/parser/Parser.java
#	src/main/java/pardiyem/task/Task.java

Nixx162 committed:
feat: Add GUI for Pardiyem

Nixx162 committed:
Merge branch 'branch-Level-10'

Nixx162 committed:
chore: Add some assertions to the code

Nixx162 committed:
refactor: make changes to improve code quality

Nixx162 committed:
Merge pull request #1 from Nixx162/branch-A-Assertions

chore: Add some assertions to the code

Nixx162 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/pardiyem/task/TaskList.java

Nixx162 committed:
Merge pull request #3 from Nixx162/branch-A-CodeQuality

refactor: make changes to improve code quality

Nixx162 committed:
docs: Create a user guide for the chatbor

Nixx162 committed:
docs: Add GFMD-styled text to README.md

Nixx162 committed:
build: add minor modiffication to the gradle config

Nixx162 committed:
build: Modify gradle build for cross-platform use

Nixx162 committed:
feat: Modify return message to be more user friendly

The initial return of just a blank string is not very helpful. It is
better to have a descriptive message like:

"There are no tasks in your current list"

This commit is made to follow that principle.

Nixx162 committed:
feat: Modify bye command to close the GUI

The previous iteration of the bye command does nothing but to make
Pardiyem say the goodbye message. The user still can chat with Pardiyem
even after that. This commit is made to remedy that, so as to stay true
with the purpose of the command.

Nixx162 committed:
docs: Add a screenshot of the UI

Nixx162 committed:
feat: Make Pardiyem greet the user

Nixx162 committed:
style: Tweak the UI a little bit for better readability

Nixx162 committed:
Merge branch 'master' of https://github.com/Nixx162/ip
